### PR TITLE
Tantivy binding

### PR DESCRIPTION
**Pull request by juchiast**
I follow the basic_search example: https://github.com/tantivy-search/tantivy/blob/master/examples/basic_search.rs

Running `cargo test bindings::tantivy::tests::test -- --nocapture`, you should see it print out:
```
{"title":["Ut enim ad minim veniam"]}
```

### CODE REVIEW

**Comment by juchiast** [comment]
@mitchtbaum 
You should add something like this one: https://github.com/jwilm/racerd/blob/master/scripts/assert_cargo_lock_unchanged to prevent uncommitted Cargo.lock . I see it happened many times.

**Comment by mitchtbaum** [comment]
Would you please also add [deleting and updating documents](https://github.com/tantivy-search/tantivy/blob/master/examples/deleting_updating_documents.rs)?

File: src/bindings/tantivy.rs [hunk 1]
```diff
@@ -0,0 +1,297 @@
+use rlua::prelude::*;
+use rlua::UserData;
+use rlua::UserDataMethods;
+
+fn combine<T: std::ops::BitOr<Output = T> + Clone>(vec: Vec<T>) -> T {
+    assert!(!vec.is_empty());
+    let mut res = vec[0].clone();
+    for i in vec {
```
Comment (fulmicoton) [inline src/bindings/tantivy.rs]: no need to clone the `Vec`.

**Comment by juchiast** [comment]
@mitchtbaum Added.

**Comment by mitchtbaum** [comment]
Do you think an update function that deletes and inserts would make sense also?

**Comment by juchiast** [comment]
Er, I don't think so.

**Comment by Arnaz87** [comment]
This error is appearing each binding you pull request, I think you are using nightly or an old version

```
error[E0658]: access to extern crates through prelude is experimental (see issue #44660)
  --> src/bindings/fs.rs:48:25
   |
48 |         let lua_value = rlua_serde::to_value(lua, &list_value)?;
   |                         ^^^^^^^^^^
```

**Comment by juchiast** [comment]
This feature is stabilized in rust 1.30, I think you are using an older rust version on you machine.

And what does the error have to do with my PR anyway? The error you pasted are in fs binding, which I do not touch.

**Comment by Arnaz87** [comment]
You are right, it was from previous, but yours still has it
~~~
error[E0658]: access to extern crates through prelude is experimental (see issue #44660)
  --> src/bindings/tantivy.rs:15:19
   |
15 | struct IntOptions(tantivy::schema::IntOptions);
   |                   ^^^^^^^
~~~

I'm gonna update rust then anyway

**Pull request merged**